components/admin_console/team_channel_settings/channel/details/channel_details.tsx

```diff
@@ -21,32 +24,44 @@ import {ChannelModes} from './channel_modes';
 import {ChannelGroups} from './channel_groups';
 import {ChannelProfile} from './channel_profile';
 
-export default class ChannelDetails extends React.Component {
-    static propTypes = {
-        channelID: PropTypes.string.isRequired,
-        channel: PropTypes.object.isRequired,
-        team: PropTypes.object.isRequired,
-        groups: PropTypes.arrayOf(PropTypes.object).isRequired,
-        totalGroups: PropTypes.number.isRequired,
-        allGroups: PropTypes.object.isRequired,
-        actions: PropTypes.shape({
-            getGroups: PropTypes.func.isRequired,
-            linkGroupSyncable: PropTypes.func.isRequired,
-            unlinkGroupSyncable: PropTypes.func.isRequired,
-            membersMinusGroupMembers: PropTypes.func.isRequired,
-            setNavigationBlocked: PropTypes.func.isRequired,
-            getChannel: PropTypes.func.isRequired,
-            getTeam: PropTypes.func.isRequired,
-            patchChannel: PropTypes.func.isRequired,
-            updateChannelPrivacy: PropTypes.func.isRequired,
-        }).isRequired,
+interface ChannelDetailsProps {
+    channelID: string;
+    channel: Channel;
+    team: Team;
+    groups: Group[];
+    totalGroups: number;
+    allGroups: {[gid: string]: Group[]};
+    actions: {
+        getGroups: (channelID: string, q?: string, page?: number, perPage?: number) => ActionFunc;
+        linkGroupSyncable: (groupID: string, syncableID: string, syncableType: SyncableType, patch: SyncablePatch) => ActionFunc;
+        unlinkGroupSyncable: (groupID: string, syncableID: string, syncableType: SyncableType) => ActionFunc;
+        membersMinusGroupMembers: (channelID: string, groupIDs: Array<string>, page?: number, perPage?: number) => ActionFunc;
+        setNavigationBlocked: (blocked: boolean) => null;
```

Comment:
not sure about the return type on this. its in admin_actions

Reply:
You can define the type inline here for now, we haven't converted that file so there's no type for it yet.